isic/zip_download/api.py

```diff
@@ -109,69 +108,38 @@ def create_zip_download_url(request: HttpRequest, payload: SearchQueryIn):
     )
 
 
-def _cloudfront_signer(message: bytes) -> bytes:
-    # See the documentation for CloudFrontSigner
-    return rsa.sign(
-        message,
-        rsa.PrivateKey.load_pkcs1(default_storage.cloudfront_key.encode("utf8")),
-        "SHA-1",
-    )
-
-
 def _zip_file_listing_generator(qs: QuerySet[Image], token: str) -> Generator[dict[str, str]]:
     def extension_from_str(s: str) -> str:
         return PurePosixPath(s).suffix.lstrip(".")
 
-    if settings.ISIC_ZIP_DOWNLOAD_WILDCARD_URLS:
-        # this is a performance optimization. repeated signing of individual urls
-        # is slow when generating large descriptors. this allows generating one signature and
-        # using it for all urls.
-        signer = CloudFrontSigner(default_storage.cloudfront_key_id, _cloudfront_signer)
-        # create a wildcard signature that allows access to * and pass this to the zipstreamer.
-        # since zip_api_auth uses a shared secret that only it and the zipstreamer know, this
-        # can't be intercepted by someone looking at the zipstreamer url.
-        bucket_url = f"https://{default_storage.custom_domain}/*"
-        policy = signer.build_policy(
-            bucket_url,
-            date_less_than=datetime.now(tz=UTC) + timedelta(days=1),
-        )
-        signed_url = signer.generate_presigned_url(bucket_url, policy=policy)
-        for image in (
-            qs.values("accession__blob", "accession__sponsored_blob", "public", "isic_id")
-            .order_by()
-            .iterator()
-        ):
-            if image["public"]:
-                # storages['sponsored'].url still goes through all of the presigning logic which
-                # significantly slows things down.
-                url = f"https://{storages['sponsored'].bucket_name}.s3.amazonaws.com/{image['accession__sponsored_blob']}"
-                zip_path = (
-                    f"{image['isic_id']}.{extension_from_str(image['accession__sponsored_blob'])}"
-                )
-            else:
-                url = signed_url.replace("*", image["accession__blob"])
-                zip_path = f"{image['isic_id']}.{extension_from_str(image['accession__blob'])}"
-
-            yield {
-                "url": url,
-                "zipPath": zip_path,
-            }
+    if hasattr(default_storage, "base_url"):
+        default_url_prefix = default_storage.base_url.rstrip("/")
+    else:
+        default_url_prefix = f"https://{default_storage.bucket_name}.s3.amazonaws.com"
```

Comment:
```suggestion
        default_url_prefix = f"https://{default_storage.bucket_name}.s3.{default_storage.region_name}.amazonaws.com"
```

You should include the region too. This will technically work, but only as a legacy compatibly: https://docs.aws.amazon.com/AmazonS3/latest/userguide/VirtualHosting.html